src/main/java/com/minecolonies/coremod/colony/requestsystem/management/manager/StandardRequestManager.java

```diff
@@ -406,71 +380,89 @@ public NBTTagCompound serializeNBT()
     @Override
     public void deserializeNBT(final NBTTagCompound nbt)
     {
-        if (playerResolver != null)
-        {
-            ResolverHandler.removeResolverInternal(this, this.playerResolver);
-        }
+        executeDeserializationStepOrMarkForUpdate(nbt,
+          NBT_VERSION,
+          NBTTagCompound::getInteger,
+          v -> version = v);
+
+        executeDeserializationStepOrMarkForUpdate(nbt,
+          NBT_DATASTORE,
+          NBTTagCompound::getCompoundTag,
+          c -> dataStoreManager = getFactoryController().deserialize(c));
+
+        executeDeserializationStepOrMarkForUpdate(nbt,
+          NBT_ID_REQUEST_IDENTITIES,
+          NBTTagCompound::getCompoundTag,
+          c -> requestIdentitiesDataStoreId = getFactoryController().deserialize(c));
+        executeDeserializationStepOrMarkForUpdate(nbt,
+          NBT_ID_REQUEST_RESOLVER_IDENTITIES,
+          NBTTagCompound::getCompoundTag,
+          c -> requestResolverIdentitiesDataStoreId = getFactoryController().deserialize(c));
+        executeDeserializationStepOrMarkForUpdate(nbt,
+          NBT_ID_PROVIDER_ASSIGNMENTS,
+          NBTTagCompound::getCompoundTag,
+          c -> providerRequestResolverAssignmentDataStoreId = getFactoryController().deserialize(c));
+        executeDeserializationStepOrMarkForUpdate(nbt,
+          NBT_ID_REQUEST_RESOLVER_ASSIGNMENTS,
+          NBTTagCompound::getCompoundTag,
+          c -> requestResolverRequestAssignmentDataStoreId = getFactoryController().deserialize(c));
+        executeDeserializationStepOrMarkForUpdate(nbt,
+          NBT_ID_REQUESTABLE_TYPE_ASSIGNMENTS,
+          NBTTagCompound::getCompoundTag,
+          c -> requestableTypeRequestResolverAssignmentDataStoreId = getFactoryController().deserialize(c));
+
+        executeDeserializationStepOrMarkForUpdate(nbt,
+          NBT_ID_PLAYER,
+          NBTTagCompound::getCompoundTag,
+          c -> playerRequestResolverId = getFactoryController().deserialize(c));
+
+        executeDeserializationStepOrMarkForUpdate(nbt,
+          NBT_ID_RETRYING,
+          NBTTagCompound::getCompoundTag,
+          c -> retryingRequestResolverId = getFactoryController().deserialize(c));
+
+        updateIfRequired();
+    }
 
-        if (retryingResolver != null)
+    private <T> void executeDeserializationStepOrMarkForUpdate(@NotNull final NBTTagCompound nbt, @NotNull final String key, @NotNull final BiFunction<NBTTagCompound, String, T> extractor, @NotNull final Consumer<T> valueConsumer)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 230 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)